lib/cartopy/mpl/solar.py

```diff
@@ -0,0 +1,124 @@
+# (C) British Crown Copyright 2018, Met Office
```

Comment:
I'm not entirely sure whether this module should be public or not. It seems like mostly internal implementation details for `ax.nightshade`. Someone who wants to calculate this stuff would probably use dedicated packages, so I'm leaning slightly toward making this private until we get a proper downstream requirement for it.

Reply:
I agree. I don’t think we want people using this functionality in CartoPy for anything other than this plot function. We can always revisit this in the future once another use case appears.

Reply:
I agree and amended the pull request and changed the module file to be: `_solar.py`, so this should be addressed now.

Reply:
GitHub seems to have changed its mind and isn't auto-hiding outdated comments if they're handled by a commit and not explicitly resolved.